trie/trieStorageManagerWithoutPruning.go

```diff
@@ -11,23 +11,23 @@ import (
 // trieStorageManagerWithoutPruning manages the storage operations of the trie, but does not prune old values
 type trieStorageManagerWithoutPruning struct {
 	common.StorageManager
-	storageManagerExtension
+	storage *trieStorageManager
 }
 
 // NewTrieStorageManagerWithoutPruning creates a new instance of trieStorageManagerWithoutPruning
-func NewTrieStorageManagerWithoutPruning(tsm common.StorageManager) (*trieStorageManagerWithoutPruning, error) {
-	if check.IfNil(tsm) {
+func NewTrieStorageManagerWithoutPruning(sm common.StorageManager) (*trieStorageManagerWithoutPruning, error) {
+	if check.IfNil(sm) {
 		return nil, ErrNilTrieStorage
 	}
 
-	sm, ok := tsm.(storageManagerExtension)
+	tsm, ok := sm.GetBaseTrieStorageManager().(*trieStorageManager)
 	if !ok {
-		return nil, errors.New("invalid storage manager type" + fmt.Sprintf("%T", tsm))
+		return nil, errors.New("invalid storage manager type" + fmt.Sprintf("%T", sm.GetBaseTrieStorageManager()))
```

Comment:
if you use `fmt.Errorf` the usage of fmt.Sprintf is no longer required

Reply:
Refactored.